File: app/src/main/java/com/gustu/logbook/fragment/home/presenter/HomePresenter.java
package com.gustu.logbook.fragment.home.presenter;

import android.util.Log;

import com.gustu.logbook.fragment.home.interfaces.HomeView;
import com.gustu.logbook.fragment.home.model.DataItem;
import com.gustu.logbook.fragment.home.model.LogbookResponse;
import com.gustu.logbook.network.BaseURLJSON;
import com.gustu.logbook.network.BaseURLXML;
import com.gustu.logbook.sharePreferences.SharedPrefUtil;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class HomePresenter {
HomeView homeView;
BaseURLJSON baseURLJSON;
final String TAG ="HOME PRESENTER";
List<DataItem> dataItemList = new ArrayList<>();
    public HomePresenter(HomeView homeView) {
        this.homeView = homeView;
        if (this.baseURLJSON ==null){
            this.baseURLJSON =new BaseURLJSON();
        }

    }
    public void getLogbook(){
        String key = SharedPrefUtil.getString("kode_pegawai");
        baseURLJSON.getAPI().getLogbook(key).enqueue(new Callback<LogbookResponse>() {
            @Override
            public void onResponse(Call<LogbookResponse> call, Response<LogbookResponse> response) {
                if (response.isSuccessful()){
                    Log.d("HOME PRESENTER", "onResponse: "+response.code());
                    LogbookResponse logbookResponse = response.body();
                    dataItemList = logbookResponse.getData();
                    homeView._onDataLoad(dataItemList);
                }
            }

            @Override
            public void onFailure(Call<LogbookResponse> call, Throwable t) {
                Log.d(TAG, "onFailure: "+t.toString());
                    homeView._onDataFail();
            }
        });
    }
}

File: app/src/main/java/com/gustu/logbook/fragment/home/presenter/HomePresenter.kt
package com.gustu.logbook.fragment.home.presenter

import android.util.Log

import com.gustu.logbook.fragment.home.interfaces.HomeView
import com.gustu.logbook.fragment.home.model.GetLogBook
import com.gustu.logbook.main.model.deleteLogbook.ResponseDeleteLogbook
import com.gustu.logbook.network.BaseURLJSON
import com.gustu.logbook.network.BaseURLXML
import com.gustu.logbook.sharePreferences.SharedPrefUtil

import java.util.ArrayList

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class HomePresenter(internal var homeView: HomeView) {
    internal var baseURLJSON: BaseURLJSON? = null
    internal var baseURLXML:BaseURLXML?=null
    internal val TAG = "HOME PRESENTER"
    internal var dataItemList: List<GetLogBook>? = ArrayList()

    init {
        if (this.baseURLJSON == null && this.baseURLXML==null) {
            this.baseURLJSON = BaseURLJSON()
            this.baseURLXML = BaseURLXML()
        }

    }

    fun getLogbook() {
        val key = SharedPrefUtil.getString("kode_pegawai")
        baseURLJSON!!.api.getLogbook(key!!).enqueue(object : Callback<List<GetLogBook>> {
            override fun onResponse(call: Call<List<GetLogBook>>, response: Response<List<GetLogBook>>) {
                if (response.isSuccessful) {
                    dataItemList = response.body()
                    homeView._onDataLoad(dataItemList)
                }
            }

            override fun onFailure(call: Call<List<GetLogBook>>, t: Throwable) {
                homeView._onDataFail()
            }
        })
    }
    fun deleteLogbook(idLogbook:String){
        baseURLXML!!.api.deleteLogbook(idLogbook).enqueue(object :Callback<ResponseDeleteLogbook>{
            override fun onResponse(call: Call<ResponseDeleteLogbook>, response: Response<ResponseDeleteLogbook>) {
                if (response.isSuccessful){
                    homeView._DataDeleted()
                }
            }

            override fun onFailure(call: Call<ResponseDeleteLogbook>, t: Throwable) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }
        })
    }
}
